src/services/exile_service.py

```diff
@@ -1,34 +1,81 @@
 import discord
 import logging
-from util import log_info_and_embed, add_and_remove_role, send_dm
-from enums import Role
+from util import log_info_and_embed, add_and_remove_role, send_dm, user_has_role
+from enums import Role, ExileStatus
+from database import users_database, exiles_database
+from typing import Optional
+import datetime
+from database.models import Exile, User
 
 logger = logging.getLogger(__name__)
 
 
 async def exile_user(
-    logging_embed: discord.Embed, user: discord.Member, duration: str, reason: str
+    logging_embed: discord.Embed,
+    user: discord.Member,
+    duration: Optional[datetime.timedelta],
+    reason: str,
 ):
-    # TODO implement this functionality
-    log_info_and_embed(
-        logging_embed, logger, f"going to exile {user.mention} for {duration}"
+    # change user role
+    await add_and_remove_role(
```

Comment:
looking at the function definition i think we'll have a bug where the user could potentially not have the role, but we're still attempting to remove it. this is a hard constraint on exile where the user needs the verified role

Reply:
I think I had just misread a request that Claire had about unexile, I agree we should move this to then end, updated

Reply:
Makes sense, I've added a check to the service and updated the command to better pass around errors